packages/eas-cli/src/commands/worker/deploy.ts

```diff
@@ -0,0 +1,193 @@
+import chalk from 'chalk';
+import fs from 'node:fs/promises';
+import * as path from 'node:path';
+
+import EasCommand from '../../commandUtils/EasCommand';
+import Log from '../../log';
+import { ora } from '../../ora';
+import { createProgressTracker } from '../../utils/progress';
+import * as WorkerAssets from '../../worker/assets';
+import { getSignedDeploymentUrlAsync } from '../../worker/deployment';
+import { UploadParams, batchUploadAsync, uploadAsync } from '../../worker/upload';
+
+const isDirectory = (directoryPath: string): Promise<boolean> =>
+  fs
+    .stat(directoryPath)
+    .then(stat => stat.isDirectory())
+    .catch(() => false);
+
+export default class WorkerDeploy extends EasCommand {
+  static override description = 'deploy an Expo web build';
+  static override aliases = ['deploy'];
+
+  // TODO(@kitten): Keep command hidden until worker deployments are live
+  static override hidden = true;
+  static override state = 'beta';
+
+  static override flags = {
+    // TODO(@kitten): Allow deployment identifier to be specified
+  };
+
+  static override contextDefinition = {
+    ...this.ContextOptions.DynamicProjectConfig,
+    ...this.ContextOptions.ProjectDir,
+    ...this.ContextOptions.LoggedIn,
+  };
+
+  async runAsync(): Promise<void> {
+    Log.warn('EAS Worker Deployments are in beta and subject to breaking changes.');
+
+    const {
+      getDynamicPrivateProjectConfigAsync,
+      loggedIn: { graphqlClient },
+    } = await this.getContextAsync(WorkerDeploy, {
+      nonInteractive: true,
+    });
+
+    const { projectId, projectDir, exp } = await getDynamicPrivateProjectConfigAsync();
```

Comment:
We can use new env vars pulled from EAS servers for config resolution here later on. I can add it in the future.

Reply:
That'd be awesome 🎉 We have yet to figure out env vars and secrets. I reckon, secrets _should_ be sourced locally, and depending on whether this is run on an EAS Custom Build step (TBD) or locally, they'd be combined with remote env vars.

The problem is that we haven't figured out yet how to source local env vars, i.e. secrets. We might have to try to crawl the Babel tree for them, or come up with a different solution that doesn't require the user to specify them explicitly in the project config

Reply:
If you have any thoughts on the flow, we've got a ticket for this now, so if you want to throw in any comments, thoughts, suggestions, an implementation path (whatever really) I hope this makes that easier 🙌 https://linear.app/expo/issue/ENG-13428/environment-variables